### PR TITLE
Update constants.js

### PR DESCRIPTION
**Pull request by gormless87**
Hi @mbystedt , I've taken a first stab at freezing the constants. Not sure about how/if to freeze REVIEWER_ENV_MAP, please advise.

### CODE REVIEW

**Comment by mbystedt** [comment]
@gormless87 Well, the first task will be to fix the error messages! You should be able to run this locally:
`npm run eslint`

The correct call is Object.freeze().
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze

**Comment by mbystedt** [comment]
Looking good!

Re: REVIEWER_ENV_MAP
Answer: A Map is not a good choice to use for a constant as there is no way for it to be constant!
https://stackoverflow.com/questions/35747325/is-there-a-way-to-freeze-an-es6-map
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map

Two options:
1. I had a little look at how it is used and it should be fairly simple to replace REVIEWER_ENV_MAP with an array of objects. Just fix the loop it is used in. Hint: Object.freeze is not recursive.
2. You can add another item to the backlog to take care of it. ;-)

**Comment by gormless87** [comment]
@mbystedt -As you have already observed I have corrected the errors. Will add backlog item to address the map to array conversion.

**Comment by gormless87** [comment]
@mbystedt - eslint ran clean locally with the changes I have made (map object changed to array of objects), have a look and let me know. How do I run linting here in GitHub?

**Comment by gormless87** [comment]
Never mind, I  see it passed linting in the 'Al checks have passed' section above.

File: lib/Jira.js [hunk 1]
```diff
@@ -65,8 +65,8 @@ class Jira {
             const branchName = this.jiraSettings.branchName
             const self = this
 
-            // loop for each env.
-            for (const [jiraTargetEnv, reviewer] of REVIEWER_ENV_MAP) {
+            // loop for each env using ARRAY.
+            for (let i = 0; i < JIRA_ENV_REVIEWER.lenght; i++) {
```
Comment (mbystedt) [inline lib/Jira.js]: Is the index needed? If not, a for-of loop eliminates the i variable and makes typos less likely. Case in point: .lenght -> .length

File: lib/constants.js [hunk 2]
```diff
@@ -22,21 +22,21 @@ const ISSUE_TRANSITION_ACTION_NAME = {
...
-const REVIEWER_ENV_MAP = new Map([
-    [JIRA_TARGET_ENV.DLVR, 'Developer'],
-    [JIRA_TARGET_ENV.TEST, 'IIT'],
-    [JIRA_TARGET_ENV.PROD, 'Business'],
-])
-const DB_ACTION = { BACKUP: 'backup', RECOVERY: 'recovery' }
+const ENV = Object.freeze({ BUILD: 'build', DEV: 'dev', DLVR: 'dlvr', TEST: 'test', PROD: 'prod' })
+const JIRA_TARGET_ENV = Object.freeze({ DLVR: 'DLVR', TEST: 'TEST', PROD: 'PROD' })
+const JIRA_ENV_REVIEWER = [
+    Object.freeze({ ENV: 'DLVR', REVIEWER: 'Developer' }),
```
Comment (mbystedt) [inline lib/constants.js]: Should use the constant for the environment.

File: lib/constants.js [hunk 3]
```diff
@@ -22,21 +22,21 @@ const ISSUE_TRANSITION_ACTION_NAME = {
...
+    Object.freeze({ ENV: 'TEST', REVIEWER: 'IIT' }),
+    Object.freeze({ ENV: 'DLVR', REVIEWER: 'Business' }),
+]
```
Comment (mbystedt) [inline lib/constants.js]: The objects in the array are frozen. Q: Is the array frozen? What would happen if I did:
JIRA_ENV_REVIEWER[0] = 'bob';

File: lib/Jira.js [hunk 4]
```diff
@@ -84,8 +84,7 @@ class Jira {
                     repoName,
                     changeBranch,
                     branchName,
-                    jiraTargetEnv,
-                    reviewer
+                    JIRA_ENV_REVIEWER[i].REVIEWER
```
Comment (mbystedt) [inline lib/Jira.js]: Seems to have accidentally replaced the environment and the reviewer with just the reviewer.

File: lib/Jira.js (hunk 1 above)
Comment (gormless87) [reply lib/Jira.js]: Corrected

File: lib/constants.js (hunk 3 above)
Comment (gormless87) [reply lib/constants.js]: corrected

File: lib/Jira.js (hunk 4 above)
Comment (gormless87) [reply lib/Jira.js]: corrected

**Comment by gormless87** [comment]
@mbystedt  - In lib/constants.js there are two objects that list environments -ENV and JRIA_TARGET_ENV. "ENV" includes four environments specified in lowercase, while "JRIA_TARGET_ENV" includes three environments and specifies the values in uppercase. I do not see why both are needed and propose removing JIRA_TARGET_ENV and using only ENV. 

const ENV = Object.freeze({ BUILD: 'build', DEV: 'dev', DLVR: 'dlvr', TEST: 'test', PROD: 'prod' })
const JIRA_TARGET_ENV = Object.freeze({ DLVR: 'DLVR', TEST: 'TEST', PROD: 'PROD' })

Any gotchas that you can see?

**Comment by mbystedt** [comment]
> @mbystedt - In lib/constants.js there are two objects that list environments -ENV and JRIA_TARGET_ENV. "ENV" includes four environments specified in lowercase, while "JRIA_TARGET_ENV" includes three environments and specifies the values in uppercase. I do not see why both are needed and propose removing JIRA_TARGET_ENV and using only ENV.
> 
> const ENV = Object.freeze({ BUILD: 'build', DEV: 'dev', DLVR: 'dlvr', TEST: 'test', PROD: 'prod' })
> const JIRA_TARGET_ENV = Object.freeze({ DLVR: 'DLVR', TEST: 'TEST', PROD: 'PROD' })
> 
> Any gotchas that you can see?

TLDR: Let's call that out of scope.

I think the idea is that it is a mapping from environment (in OpenShift?) to the same environment in JIRA. The capitalization may be important. So, the two values might be necessary. The main problems I see is with how it is coded is that it is more error prone than needed and there's no handling of unknown environments. See: transitionRfdAndSubtaksTobResolved

**Review by mbystedt — APPROVED** [review]
Looks good!

**Comment by cvarjao** [comment]
I will take the changes and test with the RFC upgrade. thanks!
